kp_test.go

```diff
@@ -1528,7 +1402,136 @@ func TestRestoreKey(t *testing.T) {
 	assert.True(t, gock.IsDone(), "Expected HTTP requests not called!")
 }
 
-// TestSetKeyPolicies tests the method SetPolicy which makes a request to Key Protect API to set Policies for a key
+// TestSetInstancePolicies tests the methods that set and get instance policies
+func TestSetAndGetInstancePolicies(t *testing.T) {
+	defer gock.Off()
+	allpolicies := []byte(`{
+		"metadata":{
+			"collectionType":"application/vnd.ibm.kms.policy+json",
+			"collectionTotal":2
+		},
+		"resources": [
+			{
+				"createdBy": "IBMid-50BE1MTM26",
+				"creationDate": "2020-04-22T15:14:29Z",
+				"lastUpdated": "2020-06-08T17:11:38Z",
+				"updatedBy": "IBMid-50BE1MTM26",
+				"policy_type": "allowedNetwork",
+				"policy_data": {
+				"enabled": true,
+				"attributes": {
+					"allowed_network": "private-only"
+				}
+				}
+			},
+			{
+				"createdBy": "IBMid-50BE1MTM26",
+				"creationDate": "2020-04-22T15:16:23Z",
+				"lastUpdated": "2020-06-08T17:11:38Z",
+				"updatedBy": "IBMid-50BE1MTM26",
+				"policy_type": "dualAuthDelete",
+				"policy_data": {
+				"enabled": true,
+				"attributes": {}
+				}
+			}
+		]
+	}`)
+	dualAuthPolicy := []byte(`{
+		"metadata":{
+			"collectionType":"application/vnd.ibm.kms.policy+json",
+			"collectionTotal":1
+		},
+		"resources": [
+			{
+				"createdBy": "IBMid-50BE1MTM26",
+				"creationDate": "2020-04-22T15:16:23Z",
+				"lastUpdated": "2020-06-08T17:11:38Z",
+				"updatedBy": "IBMid-50BE1MTM26",
+				"policy_type": "dualAuthDelete",
+				"policy_data": {
+				"enabled": false,
+				"attributes": {}
+				}
+			}
+		]
+	}`)
+	allowedNetworkPolicy := []byte(`{
+		"metadata":{
+			"collectionType":"application/vnd.ibm.kms.policy+json",
+			"collectionTotal":1
+		},
+		"resources": [
+			{
+				"createdBy": "IBMid-50BE1MTM26",
+				"creationDate": "2020-04-22T15:14:29Z",
+				"lastUpdated": "2020-06-08T17:11:38Z",
+				"updatedBy": "IBMid-50BE1MTM26",
+				"policy_type": "allowedNetwork",
+				"policy_data": {
+				"enabled": true,
+				"attributes": {
+					"allowed_network": "public-and-private"
+				}
+				}
+			}
+		]
+	}`)
+
+	gock.New("http://example.com").Reply(204)
+
+	c, _, err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+
+	err = c.SetInstancePolicies(context.Background(), true, true, true, true, "private-only")
+
+	assert.NoError(t, err)
+
+	gock.New("http://example.com").Reply(200).Body(bytes.NewReader(allpolicies))
+
+	allP, err := c.GetInstancePolicies(context.Background())
+
+	assert.NoError(t, err)
+	assert.NotNil(t, allP)
+	assert.True(t, len(allP) >= 0)
+
+	gock.New("http://example.com").Reply(204)
```

Comment:
I think you should try to match the Method and Path here as well for each request

For example
```
.Get("/instance/polices/...")
```

See the gock example here https://github.com/h2non/gock#simple-mocking-via-tests